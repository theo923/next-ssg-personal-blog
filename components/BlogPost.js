import React from 'react';
import Link from 'next/link'


const BlogPost = ({ data, httpString }) => {
    return (
        <div className='p-4 md:p-14'>
            <div className='font-prozaLibre text-5xl leading-loose'>{data?.title}</div>
            <div className='font-jost text-2xl leading-relaxed'>{data?.description}</div>
            <div className='flex my-4 font-jost text-2xl leading-relaxed'>
                {data?.tags.map(tag => <div key={tag} className="mr-2 px-3 py-1 text-sm font-bold text-gray-100 transition-colors duration-200 transform bg-gray-600 rounded cursor-pointer hover:bg-gray-500">
                    <Link href={`/category/${tag}`}>{tag}</Link>
                </div>
                )}
            </div>
            <div className='my-10' style={{left: '5%', right: '5%', width: '90%' , height: '2px', backgroundImage: 'linear-gradient(to right, transparent, rgb(48,49,51), transparent)'}}></div>
            <div className='font-hindSiliguri pb-96 mb-96 text-2xl leading-relaxed' dangerouslySetInnerHTML={{__html: httpString}}></div>
        </div>
        
    )
    
}

export default BlogPost;
