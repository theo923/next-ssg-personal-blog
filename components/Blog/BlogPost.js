import React from "react";
import Link from "next/link";
import Image from "next/image";
import Prism from "prismjs";
import "prismjs/components/prism-csharp";
import "prismjs/components/prism-jsx";
import "prismjs/components/prism-python";
import "prismjs/components/prism-java";
import SetIntersectionObserver from "../functionality/SetIntersectionObserver";
import Blind75 from "../External/Blind75";
import { useRouter } from "next/router";

const BlogPost = ({ data, httpString }) => {
  const router = useRouter();
  React.useEffect(() => {
    if (typeof window !== "undefined") {
      Prism.highlightAll(httpString);
    }
  }, []);

  React.useEffect(() => {
    if (typeof window !== "undefined") {
      Prism.highlightAll(httpString);
    }
  }, [router?.asPath]);

  return (
    <article data-test="component-blog-blogPost" className="blogPost">
      <h1>{data?.title}</h1>
      <h2>{data?.description}</h2>
      <div className="blogNodeTag my-5">
        <div>
          {data?.tags.map((tag) => (
            <button key={tag}>
              <Link href={`/category/${tag}`}>{tag}</Link>
            </button>
          ))}
        </div>
        <span className="blogPostDate">{data?.date}</span>
      </div>
      <div className="blogPostBreak" />
      {data?.tags && data?.tags.includes("Blind75") && <Blind75 />}
      <div
        className="blogPostContent"
        dangerouslySetInnerHTML={{ __html: httpString }}
      ></div>
    </article>
  );
};

export default BlogPost;
