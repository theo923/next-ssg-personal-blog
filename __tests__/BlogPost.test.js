import { shallow } from 'enzyme';
import React from 'react';
import BlogPost from '../components/Blog/BlogPost';

describe('BlogPost test', () => {
    const mockBlogPostData = {}

    it("expect BlogPost component is rendered without crashing", () => {
        const wrapper = shallow(<BlogPost mockBlogPostData={mockBlogPostData} />);   
    });

    it("expect BlogPost component is rendered && match with the Snapshot", () => {
        const wrapper = shallow(<BlogPost mockBlogPostData={mockBlogPostData} />);   
        expect(shallow(<BlogPost mockBlogPostData={mockBlogPostData} />)).toMatchSnapshot();
    });
    
});

