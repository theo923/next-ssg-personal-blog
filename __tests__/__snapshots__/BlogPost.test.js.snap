// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPost test expect BlogPost component is rendered && match with the Snapshot 1`] = `
<div
  className="p-4 md:p-14"
>
  <div
    className="font-prozaLibre text-5xl leading-loose"
  />
  <div
    className="font-spartan italic text-lg leading-relaxed"
  />
  <div
    className="flex items-center my-4 leading-relaxed"
  />
  <span
    className="font-spartan text-lg font-light text-gray-600"
  />
  <div
    className="my-10"
    style={
      Object {
        "backgroundImage": "linear-gradient(to right, transparent, rgb(48,49,51), transparent)",
        "height": "2px",
        "left": "5%",
        "right": "5%",
        "width": "90%",
      }
    }
  />
  <div
    className="font-hindSiliguri pb-96 mb-96 text-2xl leading-relaxed"
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
</div>
`;
